pkg/component/kubernetes/apiserverexposure/sni.go

```diff
@@ -142,6 +142,10 @@ func (s *sni) Deploy(ctx context.Context) error {
 		if err := managedresources.CreateForSeed(ctx, s.client, s.namespace, managedResourceName, false, serializedObjects); err != nil {
 			return err
 		}
+	} else {
+		if err := managedresources.DeleteForSeed(ctx, s.client, s.namespace, managedResourceName); err != nil {
```

Comment:
GEP-30 states:

> [...] `RemoveAPIServerProxyLegacyPort` [...]
> If enabled, gardenlet removes the old API server proxy network infrastructure on the seed side (ingress gateway port `proxy`, `Gateway`, `EnvoyFilter`, etc.)

In the current implementation, only the port and the `EnvoyFilter` are removed, but not the `Gateway`.
The `Gateway` is unusable without the port, so I'm wondering whether we should clean it up as well as described in the GEP.
WDYT?

Reply:
Sorry, I was mistaken. The `Gateway` is already removed with https://github.com/gardener/gardener/pull/11380/files#diff-0cc11d98580d91dba21a3bd4c309844de9e80091c46f4e7ce4324ecd9d0aba51R291